### DISCUSSION TITLE
Vue warnings when registering new pinia instances during testing

### DISCUSSION THREAD

**Discussion by GiovaniVercauteren** [vuejs/pinia#1234] [Help and Questions]
I am currently trying to test my store using vitest and this ends up working just fine, but the console gets absolutely flooded with warnings from Vue:

```
[Vue warn]: App already provides property with key "Symbol(pinia)". It will be overwritten with the new value.
```

with the following setup:

```js
const app = createApp({});
beforeEach(() => {
  const pinia = createPinia().use(MyPlugin);
  app.use(pinia);
  setActivePinia(pinia);
});
```

That's pretty much exactly the same as how it's described in the documentation on how to test Pinia, but I was wondering if it's expected for the console to be flooded with these warnings. Again, the end result is that my tests succeed, so everything works fine, but I'd like to not have all those warnings if at all possible.

**Comment by shuqiang0** [comment]
Maybe you executed createPinia() twice in the main.js file.

**Reply by ILoveLikeYou** [reply]
Thanks，that's how I am

**Comment by Fallsleep** [comment]
I also got this warning. I think `// you don't need to create one app per test` in the document is the problem,  create one app per test may make sense 
```typescript
beforeEach(() => {
  const app = createApp({});
  const pinia = createPinia().use(MyPlugin);
  app.use(pinia);
  setActivePinia(pinia);
});
```

**Reply by sandros06** [reply]
Do you think it's good to create an app per test ?
